### DISCUSSION TITLE
nuxi preview --port option

### DISCUSSION THREAD

**Discussion by mattgilbertnet** [nuxt/nuxt#16435] [Questions]
I'm working with `nuxt-edge` as recommended in the Nuxt Bridge migration docs. There are a couple of other discussions about production mode and SSG generally, but none seem to provide a way to run a generated site locally on a specified port.

- `server` options in `nuxt.config.js` are ignored.
- `nuxi dev` has a `--port` option, but `nuxi preview` does not.
- The port selected by `nuxi preview` changes every time the server is started, if 3000 is already being used. (Making it useless for, say, bookmarked links or links generated by another app.)
- By manually editing `.output/nitro.json`, you can add a port option to the `npx` command with something like `-l 8000`. But this file is replaced the next time `nuxi generate` is run.

Is there any way to specify a port for `nuxi preview` that I haven't tried?

**Comment by JasonLandbridge** [comment]
Adding `PORT` will change the port of the command:

`PORT=4035 nuxi preview`

![image](https://github.com/user-attachments/assets/da0abc6b-b653-4692-ac96-42870b61691a)
